### DISCUSSION TITLE
How to call a custom function with a keybinding?

### DISCUSSION THREAD

**Discussion by sirakai-pc** [qtile/qtile#2213] [Q&A]
# Issue description
I have this function
```
def togglexscreensaver(switch):
    if(switch):
        qtile.cmd_spawn('pkill disablexscreens'), #kill by pkill(killing all pids by this pattern)
    else:
        #subprocess.call([home+'/.bin/disablexscreensaver']),
        qtile.cmd_spawn("bash "+home+'/.bin/disablexscreensaver'),
    switch=not(switch),

```
`switch` is a global bool var.
This function is called trough:
```
    Key([mod], "F7", togglexscreensaver(xscreenswitch)),
#   Key([mod], "F7", lazy.function(togglexscreensaver(xscreenswitch))),
```
Which doesn't seem to work. Is there a simple and official way to do this? Because I didn't find any in the documentation.

# Qtile version
0.16.2.dev210+gd68819e6

# Stack traces

# Configuration

**Comment by elParaguayo** [comment]
Can you be more specific. What does "doesn't seem to work" mean?

What's `xscreenswitch`? If it's the global variable that you mentioned then surely there's no need to pass it as an argument.

Are you seeing any error messages in your log file?

You should be using the `lazy.function` way of doing this. The syntax is `lazy.function(function, *args, **kwargs)` so, in your case, you'd write `lazy.function(togglexscreensaver, xscreenswitch)`.

However, you should note that this will pass the `qtile` object as the first argument so your function needs to expect that as well.

I can write something in the docs for this as I agree it's not currently included (EDIT: see https://github.com/qtile/qtile/pull/2214).
